catalyst_voices/packages/internal/catalyst_voices_repositories/lib/src/user/user_repository.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_repositories/src/dto/user_dto.dart';
+import 'package:catalyst_voices_repositories/src/user/user_storage.dart';
+import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
+
+abstract interface class UserRepository {
+  factory UserRepository(
+    UserStorage storage,
+    KeychainProvider keychainProvider,
+  ) {
+    return UserRepositoryImpl(
+      storage,
+      keychainProvider,
+    );
+  }
+
+  Future<void> saveUser(User user);
+
+  Future<User> getUser();
```

Comment:
I would put `getUser()` before the `saveUser()`, usually we put read operations before save operations